HouseRules_Essentials/Rules/TileEffectDurationOverriddenRule.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace HouseRules.Essentials.Rules
+{
+    using System.Collections.Generic;
+    using Boardgame;
+    using Boardgame.Board;
+    using DataKeys;
+    using HarmonyLib;
+    using HouseRules.Types;
+
+    public sealed class TileEffectDurationOverriddenRule : Rule, IConfigWritable<Dictionary<TileEffect, int>>, IPatchable, IMultiplayerSafe
+    {
+        public override string Description => "Tile Effect durations are overridden.";
+
+        private static Dictionary<TileEffect, int> _globalAdjustments;
+        private static bool _isActivated;
+
+        private readonly Dictionary<TileEffect, int> _adjustments;
+
+
+        public TileEffectDurationOverriddenRule(Dictionary<TileEffect, int> adjustments)
+        {
+            _adjustments = adjustments;
+        }
+
+        public Dictionary<TileEffect, int> GetConfigObject() => _adjustments;
+
+        protected override void OnActivate(GameContext gameContext)
+        {
+            _globalAdjustments = _adjustments;
+            _isActivated = true;
+        }
+
+        protected override void OnDeactivate(GameContext gameContext) => _isActivated = false;
+
+        private static void Patch(Harmony harmony)
+        {
+            harmony.Patch(
+                original: AccessTools.Method(typeof(TileEffectState), "GetLayerExpireTime"),
+                prefix: new HarmonyMethod(
+                    typeof(TileEffectDurationOverriddenRule),
+                    nameof(TileEffectState_GetLayerExpireTime_Prefix)));
+        }
+
+        private static bool TileEffectState_GetLayerExpireTime_Prefix(ref TileEffect type, ref int __result)
+        {
+            if (!_isActivated)
+            {
+                return true;
+            }
+
+            if (_globalAdjustments.ContainsKey(type))
+            {
+                __result = _globalAdjustments[type];
+            }
+            else
+            {
+                __result = 9999;
+            }
```

Comment:
Is the defaulting to 9999 intentional? Or was the vision to ideally override only those effects that were specified?

Reply:
If the latter, we could prefix this block with:
```cs
if(!_globalAdjustments.ContainsKey(type)) {
    return true;
}
```

Reply:
Defaulting to 9999 was what the original code did, so I kept the same. Returning earlier would probably have been a better approach as then you'd only need to specify the overrides... Will fix.